### PR TITLE
Added tests for order search

### LINKED ISSUE #30673: Write new API test: Orders > Can search orders

### PR DESCRIPTION
**Pull request by rodelgc**
### All Submissions:
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?

### Changes proposed in this Pull Request:

Closes #30673.

### How to test the changes in this Pull Request:

1. Clone this repository and run the setup scripts for the e2e environment.
2. Run the API tests for Orders: `npm test -- --group=orders`.
3. The `Order Search API tests` should pass like so:
```
 PASS  tests/orders/order-search.test.js
  Order Search API tests
    ✓ can search by orderId (88ms)
    ✓ can search by billing first name (81ms)
    ✓ can search by billing last name (79ms)
    ✓ can search by billing company name (76ms)
    ✓ can search by billing address 1 (74ms)
    ✓ can search by billing address 2 (76ms)
    ✓ can search by billing city name (79ms)
    ✓ can search by billing post code (78ms)
    ✓ can search by billing email (85ms)
    ✓ can search by billing phone (81ms)
    ✓ can search by billing state (75ms)
    ✓ can search by shipping first name (79ms)
    ✓ can search by shipping last name (76ms)
    ✓ can search by shipping address 1 (75ms)
    ✓ can search by shipping address 2 (79ms)
    ✓ can search by shipping city (90ms)
    ✓ can search by shipping post code (85ms)
    ✓ can search by shipping state (117ms)
    ✓ can search by item name (98ms)
    ✓ can return an empty result set (95ms)

Test Suites: 1 passed, 1 of 5 total
Tests:       20 passed, 20 total
```
4. Verify through the UI or the "List all orders" endpoint that no orders were left over by the test.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Added: Order Search API tests

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by rodelgc** [comment]
I'm leaving this PR in draft mode for the meantime, as I'm expecting some merge conflicts with the [Order List PR](https://github.com/woocommerce/woocommerce/pull/31033). When that PR is merged, I'll resolve the conflicts in this branch before submitting it for review.

**Comment by rodelgc** [comment]
In this PR, I did not include these scenarios:
```
can search by billing address 1
can search by shipping address 1
can search by billing last name
can search by billing email
can search by item name
```
Because they are already covered by these tests in the [Orders Endpoint Acceptance Tests PR](https://github.com/woocommerce/woocommerce/pull/31033):
https://github.com/woocommerce/woocommerce/blob/c085afb1de6d261aa57b1fb95993f21453bd540e/packages/js/api-core-tests/tests/orders/orders.test.js#L405-L411

Ready for review now. Thanks!

File: packages/js/api-core-tests/data/shared/customer.js [hunk 1]
```diff
@@ -29,14 +29,14 @@ const customerBilling = {
 const customerShipping = {
 	first_name: 'Tim',
 	last_name: 'Clark',
-	company: 'Automattic',
+	company: 'Murphy LLC',
```
Comment (zhongruige) [inline packages/js/api-core-tests/data/shared/customer.js]: Curious on changing the values here and for the phone number, is this something we could have done in the setup of the test?

File: packages/js/api-core-tests/data/order.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ const order = {
 };
 
 const productLineItems = {
-	name: '',
+	name: 'Handmade Wooden Gloves',
```
Comment (zhongruige) [inline packages/js/api-core-tests/data/order.js]: Was this change needed?

**Review by zhongruige — COMMENTED** [review]
Overall looks good @rodelgc and these ran great for me locally:

```
 PASS  tests/orders/order-search.test.js
  Order Search API tests
    ✓ can search by orderId (182ms)
    ✓ can search by billing first name (105ms)
    ✓ can search by billing company name (90ms)
    ✓ can search by billing address 2 (98ms)
    ✓ can search by billing city name (92ms)
    ✓ can search by billing post code (96ms)
    ✓ can search by billing phone (96ms)
    ✓ can search by billing state (96ms)
    ✓ can search by shipping first name (97ms)
    ✓ can search by shipping last name (96ms)
    ✓ can search by shipping address 2 (89ms)
    ✓ can search by shipping city (88ms)
    ✓ can search by shipping post code (94ms)
    ✓ can search by shipping state (88ms)
    ✓ can return an empty result set when no matches were found (88ms)
```

Just had a question on the changes in the data for the product and customer.

File: packages/js/api-core-tests/data/order.js (hunk 2 above)
Comment (rodelgc) [reply packages/js/api-core-tests/data/order.js]: Oh, right! This change isn't needed anymore because it's already covered by the other search test (searching by `order_item_name`). I'll revert this change in my next commit.

File: packages/js/api-core-tests/data/shared/customer.js (hunk 1 above)
Comment (rodelgc) [reply packages/js/api-core-tests/data/shared/customer.js]: Yep, just realized that's the better approach. I'll do that on the next change. Thanks!

**Comment by rodelgc** [comment]
@zhongruige ready for another review!

File: packages/js/api-core-tests/data/order.js (hunk 2 above)
Comment (zhongruige) [reply packages/js/api-core-tests/data/order.js]: Thanks!

File: packages/js/api-core-tests/data/shared/customer.js (hunk 1 above)
Comment (zhongruige) [reply packages/js/api-core-tests/data/shared/customer.js]: No problem! Thanks for making this change!

**Review by zhongruige — APPROVED** [review]
Thanks for working on this @rodelgc!

These changes look good and the tests all ran and passed for me locally:

```
 PASS  tests/orders/order-search.test.js
  Order Search API tests
    ✓ can search by orderId (95ms)
    ✓ can search by billing first name (91ms)
    ✓ can search by billing company name (94ms)
    ✓ can search by billing address 2 (92ms)
    ✓ can search by billing city name (99ms)
    ✓ can search by billing post code (96ms)
    ✓ can search by billing phone (95ms)
    ✓ can search by billing state (98ms)
    ✓ can search by shipping first name (91ms)
    ✓ can search by shipping last name (92ms)
    ✓ can search by shipping address 2 (94ms)
    ✓ can search by shipping city (95ms)
    ✓ can search by shipping post code (101ms)
    ✓ can search by shipping state (97ms)
    ✓ can return an empty result set when no matches were found (90ms)
```

:shipit:
